### PR TITLE
Ce/app status staging

### PR DESCRIPTION
**Pull request by calellowitz**
@gcapalbo @snopoke 
Made a fair number of changes based on testing on staging. The big ones are allowing most of the fields to be nullable in the app status fact since we need to be able to support syncs with form submissions and vice versa, as well as apparently form submissions without app data. I also changed the user location dim query because the other one was not working as expected. Finally, the app status fact query is now 2 queries, one for the form info and one for the synclog info.

### CODE REVIEW

File: corehq/warehouse/migrations/0030_null_app.py
Comment (snopoke) [inline corehq/warehouse/migrations/0030_null_app.py]: Can we combine these 3 migrations into 1

File: corehq/warehouse/dbaccessors.py
Comment (snopoke) [inline corehq/warehouse/dbaccessors.py]: would be good to make the `doc` field deferred

File: corehq/warehouse/models/facts.py
Comment (gcapalbo) [inline corehq/warehouse/models/facts.py]: how come this is nullable? weren't we using the not null constraint here to make the process fail hard if it didn't get a match in the join to user dim?

File: corehq/warehouse/models/facts.py
Comment (gcapalbo) [inline corehq/warehouse/models/facts.py]: same question about these not null fields. will the process now fail silently if the left joins don't turn up matches, and is that ok?

File: corehq/warehouse/models/facts.py
Comment (calellowitz) [reply corehq/warehouse/models/facts.py]: There are forms with null users. Often submitted by the system. For the `FormFact` it seemed worth keeping them, in case they are needed in other reports later. They should not be used in the app status fact.

File: corehq/warehouse/models/facts.py
Comment (calellowitz) [reply corehq/warehouse/models/facts.py]: I forgot to account for users who have synced but never submitted forms. Those need to be nullable so those users will turn up in the report (and vice versa for the synclog fields i changed). Nothing should fail silently, it should just insert the null values we want

File: corehq/warehouse/migrations/0030_null_app.py
Comment (calellowitz) [reply corehq/warehouse/migrations/0030_null_app.py]: yeah, i will merge those into 1

File: corehq/apps/reports/standard/deployments.py
Comment (gcapalbo) [reply corehq/apps/reports/standard/deployments.py]: (realize you're probably still iterating on this but am guessing you'll want to remove this print statement when you're done)

**Comment by calellowitz** [comment]
@snopoke @gcapalbo final PR for the warehouse and then will deploy. I think i got to all the feedback here too, but will wait for one final approval before merging.

**Comment by calellowitz** [comment]
Yep! @NitigyaS did a lot of work to roll that out to all of our environments

**Comment by calellowitz** [comment]
pinging this again because i would love to start rollout

File: corehq/apps/reports/standard/deployments.py
Comment (snopoke) [inline corehq/apps/reports/standard/deployments.py]: Won't this fetch all the records? I think a `.count()` might be better

Comment (calellowitz) [reply corehq/apps/reports/standard/deployments.py]: my understanding was that using `len` here only fetches the rows once, and then they are used in the function call below. whereas we would still fetch all the rows below otherwise, but add an addition `COUNT` query if we use `.count()` here. i thought the rule of thumb was `.count()` if thats all you need or `len` if you were using the results otherwise. https://docs.djangoproject.com/en/2.0/ref/models/querysets/#django.db.models.query.QuerySet.count

Comment (snopoke) [reply corehq/apps/reports/standard/deployments.py]: But you aren't using all the rows, you're slicing the result set based on the pagination.

Comment (calellowitz) [reply corehq/apps/reports/standard/deployments.py]: that is a very good point. i will fix it

**Review by snopoke — APPROVED** [review]

**Pull request merged**
